_posts/2018-08-30-pair-programming.markdown

```diff
@@ -0,0 +1,184 @@
+- Two sets of keyboards, mice and monitors attached.
+- A spare computer for researching on the side.
+
+This gives you the ability to iterate and switch responsibilities at the speed of thought. If you have the resources
+and space, then setting up a place where any 2 engineers can drop a laptop down and be productive instantly will
+reduce a lot of the setup friction.
+
+### Remote
+
+The minimum you would need in this case is a video chat system like Skype, Google Hangouts or Slack Calls. Given
+that all of these support screen sharing too, then you can easily replicate two people sitting next to each other
+with one computer.
+
+It's worth noting that remote pairing requires stable internet, and with high bandwidth. We've both had situations
+in the past where we've had to abort pairing sessions due to internet un-reliability.
+
+We used to use ScreenHero a lot, to support multiple mice and keyboards, but [ScreenHero was bought by slack][sh]
+and that feature still isn't available for everyone. So, a lot of our pairing is done via screen sharing.
+
+There's a newcomer to the scene though, [VS Code's LiveShare][ls] gives you the ability to share an IDE, do voice
+chat, server ports and terminal sessions with a very low amount of setup. We wrote the initial draft of this post
+in-person, in aN Artsy meeting room writing and talking in real-time. If you've not seen Live Share, we posted a
+video of a [workshop we ran][ls-yt] at Artsy on YouTube.
+
+If you're looking to find more resources on pair programming, Joe Moore's: [Remote Pair Programming][rpp] is a great
+place to start. He's been working at Pivotal Labs for over a decade.
+
+### Attribution
+
+One way to encourage pair programming is by providing insight to others about when you're working as a pair. A great
+technique for this is to use [`git-duet`][git-duet]. Git Duet is a tool which extends git to make it easy to share
+attribution among contributors. It does this by having a central list of people' emails, you set up a pairing
+session, then use `git duet-commit` to replace `git commit` and the attribution shared.
+
+You can get started with:
+
+```sh
+$ brew tap git-duet/tap
+$ brew install git-duet
+```
+
+You'll then need to set up a `~/.git-authors` file which is a map of people's names to emails:
+
+```yml
+pairs:
+  ot: Orta Therox; orta
+  cs: Craig Spaeth; craig
+  af: Ash Furrow; ash
+  ka: Kana Abe; kana
+  md: Matt Dole;
+email:
+  domain: artsymail.com
+email_addresses:
+  md: mdole7@gmail.com
+```
+
+The format for each pair is `[tag]: [Name]; [email-prefix]`. This works in combination with the `email:domain` as
+the default email address host for someone. So it would infer that Craig Spaeth's email is `craig@artsymail.com`.
+If they've not set up that email yet, then you can use `email_addresses:` to provide overrides.
+
+Now your config is set up, you can start using it. In your terminal you can use `git duet ot af` to start a session
```

Comment:
Perspective on this post is inconsistent. Sometimes it's Orta writing, sometimes Yuki. To avoid problems with the reader not understanding who "I" or "myself" is, try to use proper nouns. So instead of "start a session with myself and Ash Furrow", why not "start a session with Orta and Ash"? Or something similarly specific. 

It's a nitpick, but I think it'd help clarify what you're communicating.

Reply:
agree - that's better

Reply:
@orta mind pairing on this? I'm not confident that I could massage the entire tone.

Reply:
Yeah, that's totally fine, let's try wrap it up Monday 🎉